evap/student/tests/test_views.py

```diff
@@ -52,6 +55,95 @@ def test_num_queries_is_constant(self):
+        included_states = [state for state in Evaluation.State if state >= Evaluation.State.APPROVED]
+
+        users = baker.make(UserProfile, _quantity=20, _bulk_create=True)
+        make_evaluation = partial(
+            baker.make,
+            Evaluation,
+            course__semester=self.semester,
+            participants=users,
+            voters=users[:10],
+            state=Evaluation.State.APPROVED,
+        )
+
+        # excluded
+        make_evaluation(is_rewarded=False)
+        make_evaluation(is_single_result=True)
+        make_evaluation(course__is_private=True)
+        make_evaluation(id=1043)
+        make_evaluation(course__type__id=1042)
+        make_evaluation(_quantity=len(excluded_states), state=iter(excluded_states))
+
+        # included
+        included_evaluations = [
+            *make_evaluation(_quantity=len(included_states), state=iter(included_states)),
+            make_evaluation(_voter_count=123, _participant_count=456),
+        ]
+
+        baker.make(VoteTimestamp, evaluation=included_evaluations[0])
+
+        expected_participants = sum(e.num_participants for e in included_evaluations)
+        expected_voters = sum(e.num_voters for e in included_evaluations)
+        expected_voter_percent = 100 * expected_voters // expected_participants
+
+        page = self.app.get(self.url, user=self.user)
+        self.assertIn("Fundraising", page)
+        self.assertIn("info_text_str", page)
+        self.assertIn(f"Last evaluation:", page)
+        self.assertIn(f"{expected_voters} submitted evaluations ({expected_voter_percent}%)", page)
+        self.assertIn("a quokka", page)
+        self.assertIn("10%", page)
+        self.assertIn("a dog", page)
+        self.assertIn("50%", page)
+
+    @override_settings(GLOBAL_EVALUATION_PROGRESS_REWARDS=[(Fraction("0.07"), "a dog")])
+    def test_global_reward_progress_edge_cases(self):
+        # no active semester
+        Semester.objects.update(is_active=False)
+        page = self.app.get(self.url, user=self.user)
+        self.assertNotIn("7%", page)
+        self.assertNotIn("a dog", page)
+
+        # no voters / participants -> possibly zero division
+        # also: no last vote timestamp
+        semester = baker.make(Semester, is_active=True)
+        page = self.app.get(self.url, user=self.user)
+        self.assertNotIn("Last evaluation:", page)
+        self.assertIn("0 submitted evaluations (0%)", page)
+        self.assertIn("7%", page)
+        self.assertIn("a dog", page)
+
+        # more voters than required for last reward
+        baker.make(
+            Evaluation,
+            course__semester=semester,
+            _voter_count=89,
+            _participant_count=97,
+            state=Evaluation.State.EVALUATED,
+        )
+        page = self.app.get(self.url, user=self.user)
+        self.assertIn("89 submitted evaluations (91%)", page)  # 91% is intentionally rounded down
```

Comment:
@janno42 the current code uses a rounded value here. Hypothetical szenario: 89 voters, 97 participants, so participation at 89/97 ~= 0.9175. The bar currently shows "92", it has the width for 92%, and it would thus look as if a 92%-goal was hit, while it technically was not yet reached. I'd want to be very careful about making a possibly wrong promise here.

I guess the production team has to decide what should happen here. If we're fine with the rounding, this test line can be adapted to pass. Otherwise, we have to fix the rounding in the template.

Reply:
It would be good to always round down. I would expect to need 90 voters to get to 92% here.
We could use a custom template filter? I don't like `widthratio` anyway, both the name and the parameters feel weird for our use case.